openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/Infrastructure.java

```diff
@@ -0,0 +1,59 @@
+package com.feedzai.openml.provider.lightgbm;
+
+import java.io.IOException;
+
+
+/**
+ * Enum that represents the infrastructure where code is running and consequent lgbm native libs locations.
+ */
+public class Infrastructure {
+
+    /**
+     * The CPU architecture used.
+     */
+    private final CpuArchitecture cpuArchitecture;
+
+    /**
+     * The libc implementation available.
+     */
+    private final LibcImplementation libcImpl;
+
+    public Infrastructure(final CpuArchitecture cpuArchitecture, final LibcImplementation libcImpl) {
+        this.cpuArchitecture = cpuArchitecture;
+        this.libcImpl = libcImpl;
+    }
+
+    @Override
+    public String toString() {
+        return "Infrastructure{" +
+                "cpuArchitecture=" + cpuArchitecture +
+                ", libcImpl=" + libcImpl +
+                '}';
+    }
+
+    /**
+     * Gets the native libraries folder name according to the cpu architecture and libc implementation.
+     *
+     * @return the native libraries folder name.
+     */
+    public String getLgbmNativeLibsFolder() throws IOException {
+
+        final String libraryPath;
```

Comment:
nitpick you don't need this variable. You can return immediately below instead of using breaks...

Reply:
It really makes the code cleaner. Thanks!